src/cephadm/cephadm

```diff
@@ -4237,6 +4237,15 @@ def command_run(ctx):
 @infer_image
 def command_shell(ctx):
     # type: (CephadmContext) -> int
+    (config, _) = get_config_and_keyring(ctx)
+    fsid_match = False
+    for c in config.split('\n'):
+        if 'fsid = '+ctx.fsid == c.strip():
+            fsid_match = True
+
+    if not fsid_match:
+        raise Error('fsid does not match ceph conf')
```

Comment:
This is the **exact** piece of code that we need in order to close https://tracker.ceph.com/issues/48368

Mind if you make this into a function and call it from check-host as well?

Reply:
i moved the code into a function but im not able to figure out a way to make it work in check-host. 

check-host is called when adding all hosts so there wont be a /etc/ceph/ceph.conf on a new hostbeing added to the cluster to check 

additionally fsid is not passed to the check_host call so theres no guaranteed fsid to check for.

there is no guarantee of a conf being in /etc/ceph on every host in the cluster unless the 'manage_etc_ceph_ceph_conf' is set.

i could check for that flag and then try to match the fsid in /etc/ceph/ceph.conf to one of the fsids in /var/lib/ceph(thats where infer_fsid is looking for a list of fsids) but im not seeing the value of having that check?